web-app/js/portal/cart/DownloadPanel.js

```diff
@@ -158,18 +158,37 @@ Portal.cart.DownloadPanel = Ext.extend(Ext.Panel, {
         setViewPortTab(TAB_INDEX_SEARCH);
     },
 
+    hasDuplicateWfsDownloadLinks: function(downloadHandlers) {
+
+        var wfsDownloads = [];
+        Ext.each(downloadHandlers, function(handler) {
+            Ext.each(handler.getDownloadOptions(), function(downloadOption) {
+                if (downloadOption.type == "WFS") {
+                    wfsDownloads.push(downloadOption.textKey)
+                }
+            });
+        });
+        return (wfsDownloads.hasDuplicates());
+    },
+
     _loadMenuItemsFromHandlers: function(processedValues, collection) {
 
         if (!processedValues.menuItems) {
             processedValues.menuItems = [];
         }
 
+        var groupedMenuItems = [];
+        var menuItemLabelsUsed = [];
         var downloadHandlers = Portal.cart.DownloadHandler.handlersForDataCollection(collection);
+        var hasDuplicateWFSDownloads = this.hasDuplicateWfsDownloadLinks(downloadHandlers);
+
```

Comment:
Can we extract a few functions out of this function, to make it a bit more readable? There are loops mixed with nested if/elses - pretty hard to understand currently.